dev/Scroller/Scroller.cpp

```diff
@@ -4851,6 +4909,38 @@ void Scroller::UpdateUnzoomedExtentAndViewport(
         UpdateScrollControllerValues(ScrollerDimension::VerticalScroll);
     }
 
+    if (horizontalViewportChanged && m_horizontalSnapPoints && m_horizontalSnapPointsNeedViewportUpdates)
+    {
+        // At least one horizontal scroll snap point is not near-aligned and is thus sensitive to the
+        // viewport width. Regenerate and set up all horizontal scroll snap points.
+        winrt::IObservableVector<winrt::ScrollSnapPointBase> horizontalSnapPoints =
```

Comment:
This is stylistic, but this is a situation where I'd use auto. Often I think people overuse auto. I like to be able to read those types! But in a situation where the type is easily visible on the Right Hand Side (something like this that's a clear cast) I think replacing the type on the Left Hand Side with auto is A-Okay. Something to noodle on, not part of a strict style guideline to my knowledge.

Reply:
Done, eventhough I don't like it.

Reply:
I appreciate your willingless to try, even though I know you don't like it :) I didn't either at first. And most of the time I still don't. But it does have its places.